src/fs/ioctl.rs

```diff
@@ -15,15 +19,21 @@ use {
 #[inline]
 #[doc(alias = "BLKSSZGET")]
 pub fn ioctl_blksszget<Fd: AsFd>(fd: Fd) -> io::Result<u32> {
-    backend::fs::syscalls::ioctl_blksszget(fd.as_fd())
+    // SAFETY: BLZSSZGET is a getter opcode that gets a u32.
+    #[allow(unsafe_code)]
+    let ctl = unsafe { ioctl::Getter::<ioctl::BadOpcode<{ c::BLKSSZGET }>, c::c_uint>::new() };
+    ioctl::ioctl(fd, ctl)
```

Comment:
This approach of factoring out the `unsafe` from the actual `ioctl` call is really nice! However I have one potential concern: could there be a problem if a kernel module gets unloaded and another gets loaded while the program is running?

If `unsafe` is on the `ioctl` call, we're effectively protected against that, because if you have an open `fd` that ought to prevent the module from being unloaded. But if we separate the `unsafe` from the `ioctl` call, then in theory there could be a point where all fds referring to a module are dropped, the module could be unloaded, another could be loaded, new fds could be created, and the same `ioctl::Getter` could be used without `unsafe`, but having a different meaning in the new module.

I'm not super familiar with either ioctls or kernel modules, so I may be missing something though.

Reply:
Ah, and a potential broader version of this concern is that with driver-specific ioctls, the safety of calling a particular ioctl depends on the dynamic type of the fd, because as I understand it, different drivers could interpret the same integer opcode as different operations.

Reply:
I've made the `ioctl` call `unsafe` for now.